### PR TITLE
[MERGE IN 1981] CRUD contribution link

### LINKED ISSUE #1923: 🚀 [Feature] Allow Admin to CRUD Contribution Links
## 🚀 Feature
As a guideline see #1906 

 * A new section in the admin interface for contribution links
 * Allow to create, list, show, edit and delete contribution links

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
- anlegen einer liste für automatische contribution mit den buttons für show, edit und delete einer automatischen contribution
- anlegen eines formulars für neue contributions (name, text, betrag, startDatum. EndDatum)

### Issues
- fixes #1923 
 
 
Bildschirmfoto vom 2022-05-30 15-36-15

Bildschirmfoto vom 2022-05-31 09-37-21

### Additional Info

merged into #1981 and developed forth there …

### CODE REVIEW

**Review by Tirokk — CHANGES_REQUESTED** [review]
Cool thing 🙏🏼🌟

The tests I do review closer when you have brought that to an end.

### Problems

If I edit the first entry:

- de: `Zyclus` is speld correctly `Zyklus` in german
- the `Zyclus` selection has no option selected, see image:

Bildschirmfoto 2022-06-08 um 11 12 29

### Suggestions

- Add a copy button to the link modal.

File: admin/src/locales/en.json [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+        "daily": "daily",
+        "hourly": "hourly",
+        "monthly": "monthly",
+        "none": "no cycle",
+        "null": "Please select a cycle",
+        "weekly": "weekly",
+        "yearly": "yearly"
+      },
+      "repetition": {
+        "null": "Please select a repetition"
```
Comment (Tirokk) [inline admin/src/locales/en.json:29]: Came in my mind now:

```suggestion
        "null": "please select a repetition"
```

File: admin/src/locales/en.json [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+    "noContributionLinks": "No contribution links have been created.",
+    "noEndDate": "No end-date",
+    "noStartDate": "No start-date",
+    "options": {
+      "cycle": {
+        "daily": "daily",
+        "hourly": "hourly",
+        "monthly": "monthly",
+        "none": "no cycle",
+        "null": "Please select a cycle",
```
Comment (Tirokk) [inline admin/src/locales/en.json:24]: Came in my mind now:

```suggestion
        "null": "please select a cycle",
```

File: admin/src/locales/de.json [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+    "clear": "Löschen",
+    "contributionLinks": "Contribution Links",
+    "create": "Anlegen",
+    "cycle": "Zyclus",
+    "endDate": "End-Datum",
+    "maximumAmount": "maximaler Betrag",
+    "memo": "Nachricht",
+    "name": "Name",
+    "newContributionLink": "Neuer Contribution Link",
+    "noContributionLinks": "Es sind keine Contribution Links angelegt.",
```
[the remark this answers is not in the document]
Comment (ogerly) [reply admin/src/locales/de.json]: changed in "**Beitragslink**"

File: admin/src/locales/de.json [hunk 4]
```diff
@@ -74,6 +106,7 @@
   "removeNotSelf": "Als Admin / Moderator kannst du dich nicht selber löschen.",
   "remove_all": "alle Nutzer entfernen",
   "save": "Speichern",
+  "selectOptions": {},
```
[the remark this answers is not in the document]
Comment (ogerly) [reply admin/src/locales/de.json:109]: remove unused selectOptions

File: admin/src/locales/en.json (hunk 2 above)
Comment (ogerly) [reply admin/src/locales/en.json:24]: changed

File: admin/src/locales/en.json (hunk 1 above)
Comment (ogerly) [reply admin/src/locales/en.json:29]: changed

File: admin/src/components/ContributionLinkForm.vue [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+        { value: null, text: this.$t('contributionLink.options.cycle.null') },
+        { value: 'none', text: this.$t('contributionLink.options.cycle.none') },
+        { value: 'hourly', text: this.$t('contributionLink.options.cycle.hourly') },
+        { value: 'daily', text: this.$t('contributionLink.options.cycle.daily') },
+        { value: 'weekly', text: this.$t('contributionLink.options.cycle.weekly') },
+        { value: 'monthly', text: this.$t('contributionLink.options.cycle.monthly') },
+        { value: 'yearly', text: this.$t('contributionLink.options.cycle.yearly') },
+      ],
+      repetition: [
+        { value: null, text: 'Please select an repetition' },
```
[the remark this answers is not in the document]
Comment (ogerly) [reply admin/src/components/ContributionLinkForm.vue:145]: changed

File: admin/src/components/ContributionLinkForm.spec.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      wrapper.vm.onSubmit()
+    })
+
+    it('function onReset', () => {
+      wrapper.vm.onReset()
+    })
+
+    it('function updateForm', () => {
+      wrapper.vm.updateForm()
+    })
```
[the remark this answers is not in the document]
Comment (ogerly) [reply admin/src/components/ContributionLinkForm.spec.js]: changed

File: admin/src/components/FigureQrCode.spec.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    beforeEach(() => {
+      wrapper = Wrapper()
+    })
+
+    it('renders the Div Element ".figure-qr-code"', () => {
+      expect(wrapper.find('div.figure-qr-code').exists()).toBeTruthy()
+    })
+
+    it('renders the Div Element "q-r-canvas"', () => {
+      expect(wrapper.find('q-r-canvas'))
```
[the remark this answers is not in the document]
Comment (ogerly) [reply admin/src/components/FigureQrCode.spec.js]: expect(wrapper.find('.canvas').exists()).toBe(true)

**Comment by ogerly** [comment]
dies ist teil von https://github.com/gradido/gradido/pull/1981 geworden

**Comment by ogerly** [comment]
is merged in master

**Pull request closed without merging**